Mathlib/Tactic/Algebraize.lean

```diff
@@ -0,0 +1,263 @@
+          let (_, mvar) ← mvarid.note nm decl.toExpr tp
+          return [mvar]
+      /- Otherwise, the attribute points to a constructor of the `Algebra` property. In this case,
+      we assume that the `RingHom` property is the last argument of the constructor (and that
+      this is all we need to supply explicitly). -/
+      else
+        let pargs := pargs.set! (n - 1) decl.toExpr
+        let val ← mkAppOptM p pargs
+        liftMetaTactic fun mvarid => do
+          let nm ← mkFreshBinderNameForTactic `algebraizeInst
+          let (_, mvar) ← mvarid.note nm val
+          return [mvar]
+    | none => return
+
+/-- Configuration for `algebraize`. -/
+structure Config where
+  /-- If true (default), the tactic will search the local context for `RingHom` properties
+    that can be converted to `Algebra` properties. -/
+  searchContext : Bool := true
+deriving Inhabited
+
+/-- Function elaborating `Algebraize.Config`. -/
+declare_config_elab elabAlgebraizeConfig Algebraize.Config
+
+end Algebraize
+
+open Algebraize Lean.Parser.Tactic
+
+/-- Tactic that, given `RingHom`s, adds the corresponding `Algebra` and (if possible)
+`IsScalarTower` instances, as well as `Algebra` corresponding to `RingHom` properties available
+as hypotheses.
+
+Example: given `f : A →+* B` and `g : B →+* C`, and `hf : f.FiniteType`, `algebraize f g` will add
+the instances `Algebra A B`, `Algebra B C`, and `Algebra.FiniteType A B`.
+
+See the `algebraize` tag for instructions on what properties can be added.
+
+The tactic also comes with a configuration option `searchContext`. If set to `true` (default), the
+tactic searches through the local context for `RingHom` properties that can be converted to
+`Algebra` properties. The macro `algebraize'` calls
+`algebraize (config := {searchContext := false})`,
+so in other words it only adds `Algebra` and `IsScalarTower` instances. -/
+syntax "algebraize" (config)? (ppSpace colGt term:max)* : tactic
+
+elab_rules : tactic
+  | `(tactic| algebraize $[$config]? $[$t:term]*) => do
+    let cfg ← elabAlgebraizeConfig (mkOptionalNode config)
+    let t ← t.mapM fun i => Term.elabTerm i none
+    -- We loop through the given terms and add algebra instances
+    for f in t do
+      let ft ← inferType f
+      match ft.getAppFn with
+      | Expr.const ``RingHom _ => addAlgebraInstanceFromRingHom f ft
+      | _ => throwError m!"{f} is not of type `RingHom`"
+    -- After having added the algebra instances we try to add scalar tower instances
+    for f in t do
+      match f.getAppFn with
+      | Expr.const ``RingHom.comp _ =>
+        try addIsScalarTowerInstanceFromRingHomComp f
+        catch _ => continue
+      | _ => continue
+
+    -- Search through the local context to find other instances of algebraize
+    if cfg.searchContext then
+      searchContext t
+
+/-- Version of `algebraize`, which only adds `Algebra` instances and `IsScalarTower` instances. -/
```

Comment:
```suggestion
/-- Version of `algebraize`, which only adds `Algebra` instances and `IsScalarTower` instances,
but not ...... -/
```

Reply:
@callesonne Please expand this docstring to explain what `algebraize'` does *not* do compared to `algebraize`.

Reply:
Yes will do, sorry I was away (and needed to prepare a talk) for a conference!

Reply:
done